digdag-standards/src/main/java/io/digdag/standards/command/ecs/DefaultEcsClient.java

```diff
@@ -224,13 +249,61 @@ public GetLogEventsResult getLog(
         if (nextToken.isPresent()) {
             request.withNextToken("f/" + nextToken.get());
         }
-        return logs.getLogEvents(request);
+        return retryOnRateLimit(() -> logs.getLogEvents(request));
     }
 
     @Override
     public void close()
             throws IOException
     {
-        client.shutdown();
+        retryOnRateLimit(() -> {
+            client.shutdown();
+            return true; //avoid restrict on generics with void
+        });
+    }
+
+    /**
+     * Retry func if ThrottlingException happen.
+     * Retry interval gradually increases with random jitter.
+     * @param func
+     * @param <T>
+     * @return
+     * @throws AmazonServiceException
+     */
+    @VisibleForTesting
+    public <T> T retryOnRateLimit(Supplier<T> func) throws AmazonServiceException
+    {
+        //ToDo  AmazonECSClient has its own retry policy mechanism. Evaluate it and consider it as replacement of this method.
+        final int baseIncrementalSecs = 10;
```

Comment:
I don't have a strong opinion on this, but I think this can be an instance variable.

Reply:
Got it. I fixed it.